### DISCUSSION TITLE
animated icons????????

### DISCUSSION THREAD

**Discussion by Jotaro-Gaming** [KadeArchive/Kade-Engine#2390] [Ideas]
when i was scrolling on the older versions of the offical fnf i saw that the icon grid used to only be 3 icons and it had an xml we can use that to make animated icons

**Comment by SuxxorState** [comment]
ive figured out how to implement this into 1.5-1.7, want me to explain how to do it later?

**Reply by SuxxorState** [reply]
alrighty, so first off is establishing a spot for your icons. they're naturally in images, but for this tutorial it's gonna be in shared/images.

the icons will work exactly like the character sprites do, needing a .xml sheet.

open up healthicon.hx. this is where most of the work will be done.
start off by clearing out these lines of code:
-this.char
-this.isPlayer
-isPlayer
-changeIcon
-the entire changeIcon function

once you're done, you should have a clean slate to put in the new code.

we're gonna make three new public variables underneath sprTracker, each looking like this:
``public var defaultIconScale:Float = 1;``
``public var iconScale:Float = 1;``
``public var iconSize:Float;``

now, under the antialiasing code (found in the "new" function), add a switch(char) function. (something like below)
``switch(char)
{
    case 'atlassux':
    default:
}``
here we can tile as many icons as we want using the case: code. i will be only tell you how to code a single case function so you have to figure the rest out.

under our character case (ex. atlassux), lets start by establishing where the icons are.
write a getSparrowAtlas function (something like this)
``frames = Paths.getSparrowAtlas('icons/atlassuxIcons', 'shared');``
atlassuxIcons is what the icon name is, whilst the icons/ before it is the folder that it is in. either of these can be whatever you so desire, icons/ isn't necessary, unless you want all icons in a folder. same with 'shared', that establishes what main folder it is in, and isnt necessary.

after that, let's define the size of the icons (or theyll be HUGE)
so underneath the previous line, write two lines:
``iconScale = 0.5;``
``defaultIconScale = 0.5;``
you can set the values to whatever your heart desires, but this is what we're going for.

after that, simply put
``antialiasing = true:``
this is so the icons dont look like doodoo fart
its time to establish the animations.
depending on whether you want winning icons or not, feel free to skip the winning animation code

add three lines of code, all looking something like this:
``animation.addByPrefix('default', "atlasNormal", 24, false, isPlayer);
``animation.addByPrefix('losing', "atlasLosing", 24, false, isPlayer);
``animation.addByPrefix('winning, "atlasWinning", 24, false, isPlayer);

replace the atlas parts with the name you chose for the animations in the .xml for the icon.

after youve done all of this, it should end up looking like this (image a)![Capture.PNG](https://user-images.githubusercontent.com/55562463/136758480-f147ca8a-0949-406f-98e5-c128afcd7b1c.PNG)

now, outside the char function, add 
``animation.play('default');``
``iconSize = width;``

and finally, go down to override function update
under super.update, add these two lines of code:
``setGraphicSize(Std.int(iconSize * iconScale));``
``updateHitbox();``

and now, save the file, for you are done.

just kidding. go to playstate.hx.
find a place in the variables to put two private ones in. they should look like this
``private var PlayerOneState:String  "default";``
`private var PlayerTwoState:String  "default";``

now, find the code in (health > 2).
we'll be replacing that and the icon code below it.
this is a big chunk of code, so I'll just let you copy this. if you dont want winning icons, replace the "winning" with "default".![tdu.PNG](https://user-images.githubusercontent.com/55562463/136760356-7d61e472-8571-4bed-b06f-4aae2288e462.PNG)
and finally (really this time), find the milf zooms. this should be at around line 4700.
underneath this, we want to put down this code.![jgdt.PNG](https://user-images.githubusercontent.com/55562463/136761006-636518a5-c2b3-4c7e-9304-95392e76f11e.PNG)![ujtyr.PNG](https://user-images.githubusercontent.com/55562463/136761014-b24c3f01-5093-4c79-b01b-328cd1956ee0.PNG)

that's it. sorry if i got lazy, its 4 in the morning and im ready to sleep.

**Comment by Jotaro-Gaming** [comment]
yes pls
